### PR TITLE
Enable nested variables in stiching

### PR DESCRIPTION
**Pull request by PascalSenn**
- Enables variables in input object :  `query {field(arg:{deep:{deep:{val:$variable}}}})}`
- Enables variables in lists object :  `query {field(arg:{arr:[{deep:{val:$variable}}}]})}`

### CODE REVIEW

**Comment by michaelstaib** [comment]
There are two snapshots tests that fail.

**Comment by michaelstaib** [comment]
also this has to be fixed on 11

File: src/Core/Language/Visitors/QuerySyntaxRewriter.cs [hunk 1]
```diff
@@ -38,6 +38,18 @@ public class QuerySyntaxRewriter<TContext>
...
+                    return RewriteObjectValue(objectValue, context);
+
+                case ListValueNode listValue:
+                    return RewriteListValue(listValue, context);
+
+                case VariableNode variable:
+                    return RewriteVariable(variable, context);
+
+                case IValueNode _:
+                    return node;
```
Comment (PascalSenn) [inline src/Core/Language/Visitors/QuerySyntaxRewriter.cs:51]: @michaelstaib  does this make sense?

**Comment by PascalSenn** [comment]
@michaelstaib  sorry forgot about `WIP:` 
Should be ready now. V11 is comming

File: src/Core/Language/Visitors/QuerySyntaxRewriter.cs (hunk 1 above)
Comment (michaelstaib) [reply src/Core/Language/Visitors/QuerySyntaxRewriter.cs:51]: yes

**Pull request merged**